PokedexMentoria/Feature/Details/DetailsViewController.swift

```diff
@@ -0,0 +1,163 @@
+        return imageView
+    }()
+    
+    private lazy var detailsPageViewController: TabBarComponent = {
+        let vc1 = PokemonInfoViewController()
+        viewModel.infoDelegate = vc1
+        let vc2 = PokemonAbilitiesViewController(abilities: [AbilityModel(ability: AbilityNameModel(name: "placeholder"), isHidden: false)])
+        viewModel.abilitiesDelegate = vc2
+        let vc3 = StatsViewController()
+        viewModel.statsDelegate = vc3
+        vc3.view.backgroundColor = .white
+        let viewsDictionary: [(String, UIViewController)] = [("Info", vc1), ("Abilities", vc2), ("Base Stats", vc3)]
+        let tabBarComponent = TabBarComponent(viewControllersWithTitle: viewsDictionary)
+        tabBarComponent.view.translatesAutoresizingMaskIntoConstraints = false
+        return tabBarComponent
+    }()
+    
+    private var detailsPageView: UIView {
+        detailsPageViewController.view
+    }
+    
+    init(viewModel: DetailsViewModel) {
+        self.viewModel = viewModel
+        
+        super.init(nibName: nil, bundle: nil)
+        viewModel.detailsDelegate = self
+    }
+    
+    @available(*, unavailable)
+    required init?(coder: NSCoder) {
+        return nil
+    }
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        makeLayout()
+    }
+    
+    private func makeLayout() {
+        view.addSubview(pokemonName)
+        view.addSubview(pokemonNumber)
+        view.addSubview(typesStackView)
+        view.addSubview(pokemonImage)
+        view.addSubview(detailsPageView)
+        
+        addChild(detailsPageViewController)
+        
+        let imageSize: CGFloat = view.bounds.size.width / 2
+        
+        NSLayoutConstraint.activate([
+            pokemonName.topAnchor.constraint(equalTo: view.safeAreaLayoutGuide.topAnchor, constant: .topSpacing),
+            pokemonName.leadingAnchor.constraint(equalTo: view.leadingAnchor, constant: .horizontalSpacing),
+            pokemonName.heightAnchor.constraint(equalToConstant: .labelHeight),
+
+            pokemonNumber.topAnchor.constraint(equalTo: view.safeAreaLayoutGuide.topAnchor, constant: .topSpacing),
+            pokemonNumber.trailingAnchor.constraint(equalTo: view.trailingAnchor, constant: -.horizontalSpacing),
+            pokemonNumber.heightAnchor.constraint(equalToConstant: .labelHeight),
+
+            typesStackView.topAnchor.constraint(equalTo: pokemonName.bottomAnchor, constant: 5),
+            typesStackView.leadingAnchor.constraint(equalTo: view.leadingAnchor, constant: .horizontalSpacing),
+            typesStackView.trailingAnchor.constraint(lessThanOrEqualTo: view.trailingAnchor, constant: -.horizontalSpacing),
+
+            pokemonImage.topAnchor.constraint(equalTo: typesStackView.bottomAnchor, constant: .labelsToViewSpacing),
+            pokemonImage.heightAnchor.constraint(equalToConstant: imageSize),
+            pokemonImage.widthAnchor.constraint(equalToConstant: imageSize),
```

Comment:
quando for assim que tem que ficar quadrado eu recomendo fixar um lado e o outro ser baseado nele
```suggestion
            pokemonImage.heightAnchor.constraint(equalToConstant: imageSize),
            pokemonImage.widthAnchor.constraint(equalTo: pokemonImage.heightAnchor),
```

Reply:
Corrigido (inclusive acho q esse comentário tá duplicado)